### ISSUE TITLE
Git histogram example not working

### ISSUE THREAD

**Issue by janosimas** [nushell/nushell#2730]
**Describe the bug**
I just started using `nushell` and it feels great!.
I went through the cookbook to check the examples but I could not get the git histogram example to work.

**To Reproduce**
1. Execute the command `git log "--pretty=format:%h(nu)%aN(nu)%s(nu)%aD" | lines | split column "(nu)" sha1 committer desc merged_at | histogram committer merger | sort-by merger | reverse` in a git repository

**Expected behavior**
I expected an histogram similar to the shown it the examples.

**Screenshots**
![image](https://user-images.githubusercontent.com/10848832/98205626-c04f8a80-1f38-11eb-8384-9d02bf149b29.png)

**Configuration (please complete the following information):**
 - OS: Windows 10 Enterprise - 1909
 - Nu version: 0.21.0
 - Optional features:

I did some tests and the histogram command seams to work fine.
if I change the histogram column parameter from 'committer' to ' sha1', it works.

There is something related to the column but I could not identify what's the issue.

**Comment by fdncred** [comment]
I've seen this before and unfortunately it's the difference between `git` on Linux and `git` on Windows. I just ran this on windows and it worked.

`git log --pretty="format:%h(nu)%aN(nu)%s(nu)%aD" | lines | split column "(nu)" sha1 committer desc merged_at | histogram committer merger | sort-by merger | reverse`

Feel free to submit a documentation PR on this issue.

**Comment by janosimas** [comment]
@fdncred It worked.
I see the only change is the position of `"`, did you test this command also on linux?

**Comment by fdncred** [comment]
@janosimas I did not however I've ran similar commands on mac and they work for a while and then data or syntax stop it from working. I haven't had the time to investigate further. I started writing a [how to here](https://gist.github.com/fdncred/caa086a07da3b3d7ee771d16999b13d5) but ran into problems and haven't had time to get back to it. I'd like to finish this and post it on nushell.sh somewhere. I'm willing to take contributions to this post :) if you'd like to work on it with me.

**Comment by janosimas** [comment]
Sure, I'll take a look during the weekend. :-)

**Comment by janosimas** [comment]
I did a few tests and I believe the issue is the `"`:
 - All my tests are on windows.
 - Looks like the `"format:"` in the mask is unnecessary

We don't really need the `"`, the command `git log --pretty=%h(nu)%aN(nu)%s(nu)%aD` works even better.
Using quotes added quotes to the result and interfered with the date parsing in your example.

I did the test with other terminals and using `()` causes issues and we would need to scape but with `Nu` this is not a problem.

I executed your example without the quotes and it worked perfectly:
```
git log --pretty=%h»¦«%s»¦«%aN»¦«%aE»¦«%aD | lines | split column "»¦«" commit subject name email date | update date { get date | str to-datetime} | where date < 365day | group-by name | pivot | sort-by Column1
```

**Comment by fdncred** [comment]
on my mac, this doesn't work 
`git log --pretty=%h»¦«%s»¦«%aN»¦«%aE»¦«%aD | lines | split column "»¦«" commit subject name email date | update date { get date | str to-datetime} | where date < 365day | group-by name | pivot | sort-by Column1`

nor does this statement
`git log --pretty=format:%h»¦«%s»¦«%aN»¦«%aE»¦«%aD -n 25 | lines | split column "»¦«" commit subject name email date | update date { get date | str to-datetime} | group-by date`

**Comment by janosimas** [comment]
What's the output of:
```
git log --pretty=%h»¦«%s»¦«%aN»¦«%aE»¦«%aD
```

**Comment by fdncred** [comment]
that works but the problem is somewhere along the pipeline the `date` data goes missing which causes the pipeline to fail. I'm not sure whether this is a nushell issue or a git issue.

**Comment by fdncred** [comment]
here's the real problem. for some reason, you can see there are lines with no commit.

```
git log --pretty=format:%h»¦«%s»¦«%aN»¦«%aE»¦«%aD -n 25 | lines | split column "»¦«" commit subject name email date
╭───┬──────────┬────────────────────────────┬─────────────────┬───────────────────────────────────────┬────────────────────────────────╮
│ # │ commit   │ subject                    │ name            │ email                                 │ date                           │
├───┼──────────┼────────────────────────────┼─────────────────┼───────────────────────────────────────┼────────────────────────────────┤
│ 0 │ 0ee054b1 │ Fix `to md` errors (#2729) │ Joseph T. Lyons │ JosephTLyons@users.noreply.github.com │ Fri, 6 Nov 2020 12:40:53 -0500 │
╰───┴──────────┴────────────────────────────┴─────────────────┴───────────────────────────────────────┴────────────────────────────────╯
╭───┬────────╮
│ # │ commit │
├───┼────────┤
│ 1 │        │
╰───┴────────╯
╭───┬──────────┬────────────────────────────────────────────────┬───────────────┬──────────────────────┬────────────────────────────────╮
│ # │ commit   │ subject                                        │ name          │ email                │ date                           │
├───┼──────────┼────────────────────────────────────────────────┼───────────────┼──────────────────────┼────────────────────────────────┤
│ 2 │ 80b39454 │ Change Nu Shell and NuShell to Nushell (#2728) │ Leonhard Kipp │ leonhard.kipp@web.de │ Fri, 6 Nov 2020 18:39:49 +0100 │
╰───┴──────────┴────────────────────────────────────────────────┴───────────────┴──────────────────────┴────────────────────────────────╯
╭───┬────────╮
│ # │ commit │
├───┼────────┤
│ 3 │        │
╰───┴────────╯
```

without the split column. It's like there are extra line feeds sometimes.
```
git log --pretty=format:%h»¦«%s»¦«%aN»¦«%aE»¦«%aD -n 25 | lines
╭────┬─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│  0 │ 0ee054b1»¦«Fix `to md` errors (#2729)»¦«Joseph T. Lyons»¦«JosephTLyons@users.noreply.github.com»¦«Fri, 6 Nov 2020 12:40:53 -0500            │
│  1 │                                                                                                                                             │
│  2 │ 80b39454»¦«Change Nu Shell and NuShell to Nushell (#2728)»¦«Leonhard Kipp»¦«leonhard.kipp@web.de»¦«Fri, 6 Nov 2020 18:39:49 +0100           │
│  3 │                                                                                                                                             │
│  4 │ 97f3671e»¦«web scraping with css selectors (#2725)»¦«Darren Schroeder»¦«343840+fdncred@users.noreply.github.com»¦«Tue, 3 Nov 2020 15:46:42  │
│    │ -0600                                                                                                                                       │
│  5 │                                                                                                                                             │
│  6 │ b674cee9»¦«Remove the recursely-dep'd tests (#2727)»¦«Jonathan Turner»¦«jonathandturner@users.noreply.github.com»¦«Wed, 4 Nov 2020 09:26:07 │
│    │ +1300                                                                                                                                       │
│  7 │                                                                                                                                             │
│  8 │ cb8491cf»¦«Bump to 0.22 (#2726)»¦«Jonathan Turner»¦«jonathandturner@users.noreply.github.com»¦«Wed, 4 Nov 2020 07:31:41 +1300               │
```

**Comment by janosimas** [comment]
I get the same issue with I use the `format:`, here are my two outputs:
```
git log --pretty=format:%h»¦«%s»¦«%aN»¦«%aE»¦«%aD -n 10 | lines | split column "»¦«" commit subject name email date
───┬─────────┬────────────────────────────┬───────────┬─────────────────────┬─────────────────────────────────
 # │ commit  │          subject           │   name    │        email        │              date
───┼─────────┼────────────────────────────┼───────────┼─────────────────────┼─────────────────────────────────
 0 │ fb0c3b6 │ add .env file to gitignore │ janosimas │ janosimas@gmail.com │ Tue, 15 Sep 2020 19:26:48 +0200 
───┴─────────┴────────────────────────────┴───────────┴─────────────────────┴─────────────────────────────────
───┬────────
 # │ commit
───┼────────
 1 │
───┴────────
───┬─────────┬─────────────────┬────────────┬────────────────────────────────────┬────────────────────────────────
 # │ commit  │     subject     │    name    │               email                │              date
───┼─────────┼─────────────────┼────────────┼────────────────────────────────────┼────────────────────────────────
 2 │ 88f7e32 │ Create rust.yml │ Jano Simas │ janosimas@users.noreply.github.com │ Tue, 1 Sep 2020 22:07:15 +0200
───┴─────────┴─────────────────┴────────────┴────────────────────────────────────┴────────────────────────────────
───┬────────
 # │ commit
───┼────────
 3 │
───┴────────
───┬─────────┬────────────────┬───────────────┬───────────────────┬────────────────────────────────
 # │ commit  │    subject     │     name      │       email       │              date
───┼─────────┼────────────────┼───────────────┼───────────────────┼────────────────────────────────
 4 │ 745f405 │ Create LICENSE │ Aurélien Hamy │ aunetx@yandex.com │ Tue, 1 Sep 2020 23:19:44 +0200
───┴─────────┴────────────────┴───────────────┴───────────────────┴────────────────────────────────
───┬────────
 # │ commit
───┼────────
 5 │
───┴────────
───┬─────────┬───────────────────┬───────────┬─────────────────────┬─────────────────────────────────
 # │ commit  │      subject      │   name    │        email        │              date
───┼─────────┼───────────────────┼───────────┼─────────────────────┼─────────────────────────────────
 6 │ 186675f │ Move to async API │ janosimas │ janosimas@gmail.com │ Mon, 31 Aug 2020 22:36:19 +0200
───┴─────────┴───────────────────┴───────────┴─────────────────────┴─────────────────────────────────
───┬────────
 # │ commit
───┼────────
 7 │
───┴────────
────┬─────────┬──────────────────────────────────────────────┬───────────┬─────────────────────┬─────────────────────────────────
 #  │ commit  │                   subject                    │   name    │        email        │              date
────┼─────────┼──────────────────────────────────────────────┼───────────┼─────────────────────┼─────────────────────────────────
  8 │ d0bc141 │ Automatic language detection in GoogleV2 API │ janosimas │ janosimas@gmail.com │ Sun, 30 Aug 2020 14:41:53 +0200
  9 │ 3571d04 │ Update readme                                │ janosimas │ janosimas@gmail.com │ Sun, 30 Aug 2020 12:02:28 +0200
 10 │ b712b49 │ Add documentation and better error handling  │ janosimas │ janosimas@gmail.com │ Sun, 30 Aug 2020 11:57:13 +0200
 11 │ a816d5a │ remove unimplemented google v3               │ janosimas │ janosimas@gmail.com │ Fri, 28 Aug 2020 19:50:34 +0200
 12 │ d10db56 │ Detect language with google api              │ janosimas │ janosimas@gmail.com │ Fri, 28 Aug 2020 19:49:25 +0200 
 13 │ 434e18b │ Rename files and classes                     │ janosimas │ janosimas@gmail.com │ Fri, 28 Aug 2020 19:26:48 +0200
────┴─────────┴──────────────────────────────────────────────┴───────────┴─────────────────────┴─────────────────────────────────
```

```
git log --pretty=%h»¦«%s»¦«%aN»¦«%aE»¦«%aD -n 10 | lines | split column "»¦«" commit subject name email date
───┬─────────┬──────────────────────────────────────────────┬───────────────┬────────────────────────────────────┬─────────────────────────────────
 # │ commit  │                   subject                    │     name      │               email                │              date
───┼─────────┼──────────────────────────────────────────────┼───────────────┼────────────────────────────────────┼─────────────────────────────────
 0 │ fb0c3b6 │ add .env file to gitignore                   │ janosimas     │ janosimas@gmail.com                │ Tue, 15 Sep 2020 19:26:48 +0200
 1 │ 88f7e32 │ Create rust.yml                              │ Jano Simas    │ janosimas@users.noreply.github.com │ Tue, 1 Sep 2020 22:07:15 +0200
 2 │ 745f405 │ Create LICENSE                               │ Aurélien Hamy │ aunetx@yandex.com                  │ Tue, 1 Sep 2020 23:19:44 +0200
 3 │ 186675f │ Move to async API                            │ janosimas     │ janosimas@gmail.com                │ Mon, 31 Aug 2020 22:36:19 +0200
 4 │ d0bc141 │ Automatic language detection in GoogleV2 API │ janosimas     │ janosimas@gmail.com                │ Sun, 30 Aug 2020 14:41:53 +0200
 5 │ 3571d04 │ Update readme                                │ janosimas     │ janosimas@gmail.com                │ Sun, 30 Aug 2020 12:02:28 +0200
 6 │ b712b49 │ Add documentation and better error handling  │ janosimas     │ janosimas@gmail.com                │ Sun, 30 Aug 2020 11:57:13 +0200
 7 │ a816d5a │ remove unimplemented google v3               │ janosimas     │ janosimas@gmail.com                │ Fri, 28 Aug 2020 19:50:34 +0200
 8 │ d10db56 │ Detect language with google api              │ janosimas     │ janosimas@gmail.com                │ Fri, 28 Aug 2020 19:49:25 +0200
 9 │ 434e18b │ Rename files and classes                     │ janosimas     │ janosimas@gmail.com                │ Fri, 28 Aug 2020 19:26:48 +0200
───┴─────────┴──────────────────────────────────────────────┴───────────────┴────────────────────────────────────┴─────────────────────────────────
```

**Comment by fdncred** [comment]
oh, wait! I totally missed your point. You're saying use `--pretty` without `format`. Duh. Sorry I missed that. Let me try again.

**Comment by fdncred** [comment]
darn, it still fails here
`git log --pretty=%h»¦«%s»¦«%aN»¦«%aE»¦«%aD -n 25 | lines | split column "»¦«" commit subject name email date | update date { get date | str to-datetime} | group-by date`
```
error: Type Error
  ┌─ shell:1:1
  │
1 │ git log --pretty=%h»¦«%s»¦«%aN»¦«%aE»¦«%aD -n 100 | lines | split column "»¦«" commit subject name email date | update date { get date | str to-datetime} | group-by date -f '%d/%m/%Y'
  │ ^ Expected date, found row
```

**Comment by fdncred** [comment]
Duh, i forgot there's a special `group-by` for dates.

`git log --pretty=%h»¦«%s»¦«%aN»¦«%aE»¦«%aD -n 25 | lines | split column "»¦«" commit subject name email date | update date { get date | str to-datetime} | group-by date date | pivot`
```
╭───┬────────────┬────────────────╮
│ # │ Column0    │ Column1        │
├───┼────────────┼────────────────┤
│ 0 │ 2020-11-06 │ [table 2 rows] │
│ 1 │ 2020-11-03 │ [table 4 rows] │
│ 2 │ 2020-11-02 │ [table 1 rows] │
│ 3 │ 2020-10-30 │ [table 4 rows] │
│ 4 │ 2020-10-29 │ [table 2 rows] │
│ 5 │ 2020-10-28 │ [table 2 rows] │
│ 6 │ 2020-10-27 │ [table 3 rows] │
│ 7 │ 2020-10-26 │ [table 4 rows] │
│ 8 │ 2020-10-22 │ [table 3 rows] │
╰───┴────────────┴────────────────╯
```

`group-by date` is a subcommand and it's specifically for dates. So, now everything seems to be working.

Any thoughts of what else we should add to this?
https://gist.github.com/fdncred/caa086a07da3b3d7ee771d16999b13d5

I need to go over it again for flow because i jumped back and forth through it and probably lose the readers train of thought.

**Comment by fdncred** [comment]
posted here https://www.nushell.sh/cookbook/

**Comment by fdncred** [comment]
Should we close this issue?

This works in Windows.
`git log --pretty=%h(nu)%aN(nu)%s(nu)%aD | lines | split column "(nu)" sha1 committer desc merged_at | histogram committer merger | sort-by merger |
reverse`

This works in Linux.
`git log --pretty="%h(nu)%aN(nu)%s(nu)%aD" | lines | split column "(nu)" sha1 committer desc merged_at | histogram committer merger | sort-by merger |
reverse`

The difference being, Linux requires quotes around the string to escape the parenthesis.

**Comment by janosimas** [comment]
I think it would be better to not use the parenthesis and have the same command for every system.
I like your special token `»¦«` , it could also be used in the other page.

I can do a pull request with the change in the weekend.

**Comment by fdncred** [comment]
sounds good. i'm closing this then. thanks for your help.
